### ISSUE TITLE
Code of Conduct

### ISSUE THREAD

**Issue by fluffyfreak** [pioneerspacesim/pioneer#4460]
Could we do with a Code of Conduct? Wouldn't hurt would it?

So after little research it seems that the standard from which many others are derived is the first in the list:
- https://geekfeminism.org/about/code-of-conduct/
- https://www.kernel.org/doc/html/latest/process/code-of-conduct.html
- https://thoughtbot.com/open-source-code-of-conduct
- https://www.contributor-covenant.org/
- https://www.greaterthancode.com/code-of-conduct/

Suggestions, preferences and any customisations/alterations?

**Comment by Web-eWorks** [comment]
The Contributor Covenant, or at least the latter half of it, looks like a good base to start from. It will likely require some clarification regarding the punitive process (e.g. strikes? appeals?), and a rewritten first two sections, but is otherwise sound.

**Comment by macksting** [comment]
I'm definitely still collecting my thoughts on this, and that's a lot more reading than I can do at this moment. But if this becomes a thing, I want it to be a thing LjL can live with, something that's not paralyzing.
"Project maintainers are responsible for clarifying the standards of acceptable behavior"
We have bandied about possibly adding,
"If the project maintainers become the problem, everybody should jump ship." It's awfully informal, but that's kind of one of its charms as a clause.

I'm sorry if this comment is divisive, but seriously, I have a big request here. Please don't call this "SJW crap" or any variation thereof. I'm very ambivalent on the point of a CoC, feeling both for and against it, but to write off something that's important to me _because_ it's important to me is rather alarming.

**Comment by nozmajner** [comment]
Did something happen that necessitates an expressed code of conduct more detailed than "be decent people" while I wasn't around?

**Comment by impaktor** [comment]
@nozmajner, no, no problem, we have new good contributors, doing fine work, but mean while, in the larger Linux-kernel world, here's a summary (10 minutes, by Tim Pool): https://www.youtube.com/watch?v=v5VvJiNUCIA

As for this PR: I'm strongly against:

1. We have managed very fine without a CoC so far, and the few assholes we've encountered would not in any way have been hampered by a CoC. In fact, they would _use_ the CoC and claim they've been "harassed" as we've already seen claims of that from a previous troll, who didn't tolerate the most innocent questions regarding their code.

2. We don't need a CoC to motivate why we kick someone out, if it ever would come to that. And they wouldn't feel any less wronged, as they wouldn't feel that they have violated the CoC. No one thinks themselves a crook.

3. And even if we do have a CoC, who interprets what exactly is "harassment"? Or what is "unacceptable behavior"? It's just as undefined as without a CoC, only, with a CoC we add the _illusion_ of it being defined, and us being righteous in kicking someone out. I'll bet me and @laarmen have thrown jokes at each other that could get us both kicked out, depending on who interprets the CoC. I've seen countless examples elsewhere of people claiming to be "harassed" when they receive criticism over something they've done/said. It all comes down to interpretation.

4. Do we want these alleged contributors who would need a CoC to understand how to behave? If you don't know how to be civilized with or without a CoC, then I'd rather know from the git-go (pun intended?).

5. Do we really think there are people who are assholes without a CoC but civilized with one?

6. Rather, the existence of a CoC indicates to me that this is a dysfunctioning community with bickering and harsh words thrown (such as the Linux kernel community), and I'd think twice before joining. Thankfully, pioneer has not been that at all.

7. CoCs get really absurd (such a the Linux Foundation's) when they start to enumerate things you're not allowed to do, like "make fun of someones weight". Fine, but does that mean we can make fun of someones nose? How about height? Eye color? Accent? Spelling? Typos? Cultural cuisine? (I _know_ I must have made snail-eating jokes) Or why else did they start listing these things?

8. The Linux Foundation's CoC also contains highly politicized statements, typical of ideologically possessed people. I want Pioneer to remain un-political. And no, I don't want the CoC to talk about being "inclusive", as this is New Speak for _exact_ the opposite: "you don't think exactly like me in X,Y or Z, then we will not tolerate you!". I have hundreds of examples like that right up my sleeve, and I don't want to see that in pioneer. If I see the word "inclusive" I know I'm possibly dealing with a person who sees themselves as a pinnacle of moral righteousness and that they have the duty, to "punch a nazi" (that was a thing a few months ago, if you missed it), and here's the kicker: _I_ get to decide who's "a nazi", and it's anyone I don't like.

_(semi-related: I recommend the book [The Righteous Mind: Why Good People Are Divided by Politics and Religion](https://www.goodreads.com/book/show/11324722-the-righteous-mind))_

I realize I might come off as an "asshole" to some, I've seen almost daily how the people calling for "inclusion" and "tolerance" spread bottomless hate for people who's views they don't want to understand, because it keeps them selves protected. It's the ultimate ego-trip for the Cain-like element of the human spirit: be resentful, arrogant, and deceitful, and look virtuous while doing so, at no cost to yourself.

## Alternative to a CoC

Keep status quo, where pioneer is a 100% un-political project. We who are active in the dev team are all reasonable normal people, in my opinion, and I think everyone agrees there, and I don't see any problem in keeping it that.

If someone is making life a pain, we kick them out. 100% subjective, (as it still would be under the guise of a CoC).

With that said, no hard feelings, this rant isn't aimed at any specific person.

**Comment by LuccoJ** [comment]
> Did something happen that necessitates an expressed code of conduct more detailed than "be decent people" while I wasn't around?

@nozmajner it's pretty much, I made a passing remark on IRC like "do you have a CoC yet?" that was instigated by reading about what had been going on at Linux, and which was meant mostly as somewhere in between humor and "let's see what the general feeling about that is around these part", but instead my challenge has been (semi-?)seriously taken up, and then I wished it could eat back my words.

If it was meant as a lesson in shutting up when I have no real reason to start talking about something, I must admit it was effective.

**Comment by nozmajner** [comment]
Yeah, "Be $ decent people" should be more than enough for our scale, I'd guess. Maybe with a $=swear thrown in for good measure. :)

**Comment by fluffyfreak** [comment]
@nozmajner 
> "Be decent people you bunch of fuckers"?

Yeah I see no immediate need for a CoC but then I'm on the inside looking out. 
It would be interesting to get an outside perspective on all of this somehow to see what might deter someone else from joining our community.

**Comment by macksting** [comment]
As somebody who's on the periphery, I found the extremely negative reaction far more frightening than the lack of the CoC itself. These things ostensibly exist to protect _me,_ and not being sure I need the protection is one thing, but I became a whole lot less comfortable when it got called SJW bullshit. Suddenly I wasn't sure I was safe describing why I was ambivalent about the matter. (I guess I'm not really out in a lot of places.)

**Comment by nozmajner** [comment]
@fluffyfreak Something along those lines :D

**Comment by impaktor** [comment]
> As somebody who's on the periphery, I found the extremely negative reaction far more frightening than the lack of the CoC itself. 

Well, I was the only one who reacted strongly negatively, so it's apt for me to reply: if you want to understand _why_ I reacted that way, please see Tim Pool's video on exactly how CoC is being used in the Linux Community: https://www.youtube.com/watch?v=v5VvJiNUCIA

Namely: witch hunts, mob-bullying, kicking good contributors out, harassing people, and people accusing each other of "violating the CoC", and digging up things in someones past, to get them thrown out.

> These things ostensibly exist to protect _me,_

I believe, for above stated reasons, it does nothing of the sort, but rather, it has the potential to do the opposite.

> I became a whole lot less comfortable when it got called SJW bullshit. Suddenly I wasn't sure I was safe describing why I was ambivalent about the matter

No one should be "safe" from criticism, all ideas must be allowed to be questioned. In my opinion, we've (almost) never had any problem with people's conduct, but I assume all grown up must be able to tolerate critique.

I view the whole CoC thing as a highly political, and this will then take us into the realm of politics, that I'd rather avoid, but SJW is a cancer of the mind, in my opinion, for rallying up lazy thinking mobs to go on witch hunts to ruin people, only to make themselves look more virtuous. I recommend reading the article I linked above (again here: [I Was the Mob Until the Mob Came for Me](https://quillette.com/2018/07/14/i-was-the-mob-until-the-mob-came-for-me/))

That being said, I have nothing against you or anyone else in the community, please feel free to discuss and say what ever you want, but I can not promise I will agree with you, so in that way, the world isn't safe. We get our views challenged on a regular basis, and learn new things in the process of adopting new views and more knowledge and understanding. Hopefully.

**Comment by fluffyfreak** [comment]
@macksting I'm perfectly happy being a SJW 😄 I want everything to be nice for everyone, even @impaktor 😛

**Comment by laarmen** [comment]
I wrote a lengthy thing about CoCs and why they're a good thing, but in the
end, here's my proposal for one :

"Don't be a dick[0], the definition of which falls to whoever is in
charge". At least that's clear, right? Not especially political, the
arbitrary part (which is inevitable) is explicit.

Also, SJW shouldn't be used as an insult, or a derogatory acronym. Voltaire
was a SJW, FFS!

[0] Except for @impaktor, he gets a pass 'cause he's funny sometimes.

**Comment by macksting** [comment]
@impaktor I didn't call you out directly because I wasn't sure it was dirty laundry you'd want aired, but I'll be less passive aggressive about it now that I'm able to be more direct. Thanks for coming forward to discuss it this way.
I'm all for reasoned criticism. I'll strongly consider going over the materials availed to me here when I have more brainogens to go around. I'm personally already pretty familiar with how this sort of thing can be misused or go disastrously wrong.
I was in an organization which tried to be inclusive as hell, really did, and had the size and complexity to warrant actual systems for addressing problems. Unfortunately, it was implemented poorly, and I got thrown under the bus in a grand misunderstanding created by them failing to make accommodations for my mental disability. Believe me, the language requiring those accommodations was present and not adhered to correctly, simply because _everybody_ involved lacked the expertise to actually come up with relevant policy. **_THE MORE YOU KNOW *_** 
And yeah, somebody in there may have been a Bad Actor™. She quit after I did. So there's a very real possibility my peace of mind was wrecked because somebody who shouldn't have been in charge was in charge.
Honestly, we're all really just trying to do the best we can. Ambiguity is inevitable and we must more or less learn to find common ground where we can, and ultimately we are condemned to be free and all that jazz. There's nothing for it.
Just... please be careful about things that sound like personal attacks on folks for what's basically just an attempt to do the right thing. That goes both ways, I know. I really do trust you mean well, and that you're just trying to avoid problems, and I'm all for that. And a COC can help avoid problems, or it can be used to create them. It's just a tool, a technique, and technology can't solve social problems, etc., but it can sure introduce new and interesting ones.
I'm sorry this is going on so long. I don't ask that I be immune to criticism. It just scares me more than perhaps it should to have something which ostensibly exists to allow me to safely engage be savagely dismissed with a few words that sound like every last person trying to make the world a better place is some kind of monster.

I offer two little possible additions to what might be merely a one or two line COC as it stands.
1) Until further notice, "don't be a dick" means think twice before using buzzwords.
2) If the captain gets drunk and the ship wrecks, evacuate. Either it can be salvaged later or it will sink; either way, tend to your wounded, then see what can be done for it later.

Right now the captain's sober. The scale of this project seems like it's a great scale for a benevolent dictatorship or pure anarchy. I want to stress that, personally, as a nonbinary pansexual millenial with a freaking tumblr account, I'm perfectly fine with that. This whole conversation is probably a distraction from important work, and I'm here to reproduce bugs, not build a revolution.
But please... be gentle. I've had a hell of a year.

**Comment by LuccoJ** [comment]
>  These things ostensibly exist to protect *me* [...] I became a whole lot less comfortable when it got called SJW bullshit. Suddenly I wasn't sure I was safe describing why I was ambivalent about the matter. (I guess I'm not really out in a lot of places.)

You know, I'm not really out in a lot of places either, but the very reason I am wary of these CoCs sprouting out of nowhere in so many projects, almost all inspired by roughly the same document, is that while I am gay, I absolutely do not identify with

* having stringent and specific "bullet points" with behavioral requirements that get you out of a project if you don't respect them
* having a code of conduct for an open-source project begin not with something about the project, its source status, its contributors' mindset, and its philosophy, but about how inclusive you need to be towards insert-specific-list-of-people-here; it is really the right place for that?
* ending up antagonizing a specific individual for purported "CoC breaches", as I realize that to some people, it's only a CoC breach when they see it as one, but if they breach it against you, by "outing" you for instance, then it can be fine to them, because hey, you were the one who breached the CoC in the first place (in their eyes), so it's okay (in their eyes) to fight back — I have seen this happen, and I have seen this type of CoC making defending oneself against this pattern very hard.

So in short, I feel that even though the purported goal of "these things" is to protect me, their real goal, no, at least their real *effect*, is quite far from it, and actually, it can easily be far closer to kickstarting the persecution of individuals, similarly to the kind I don't want to be subjected to.

All that glitters is not gold.

**Comment by macksting** [comment]
I'm not wildly in favor of the notion myself, which does mean two people who would be directly impacted by the relevant policy aren't particularly interested in implementing it. It hasn't been up long, so we could kick the ball down the field, which was sort of my default answer until things got... a little stressful all of a sudden. (After all, many feature requests, bugs and issues don't get addressed for quite a while if they're not pressing!) I'm leaning toward "best not," personally.

**Comment by fluffyfreak** [comment]
So general consensus is that these Code of Conduct are generally counterproductive, the two people who are willing to say that they might most be affected (_though of course there might be silent others_) do not feel that it's a good idea.

The overall feeling seems to be a no.

I'm surprised having not really had to deal with CoC outside of professional employment. Certainly didn't expect that it would get people so riled up, naive of me. 
I just posted it up expecting a slow discussion over a few weeks.

Closing as kick-it-down-the-line / won't fix.

**Comment by impaktor** [comment]
Anyone seen the CoC of SQLite? I thought it was a bit funny, in its irony:
https://sqlite.org/codeofconduct.html
